tests/property/ic_api/canister_version/test/tests.ts

```diff
@@ -0,0 +1,129 @@
+import {
+    defaultPropTestParams,
+    expect,
+    getCanisterActor,
+    it,
+    Test
+} from 'azle/test';
+import { execSync } from 'child_process';
+import fc from 'fast-check';
+
+import { _SERVICE as Actor } from './dfx_generated/canister/canister.did';
+
+export function getTests(): Test {
+    return () => {
+        it('gets the canister version from the canister', async () => {
+            await fc.assert(
+                fc.asyncProperty(
+                    fc.nat({
+                        max: 10
+                    }),
+                    async (timesToDeploy) => {
+                        // Stop and delete the canister before every run
+                        execSync(`dfx canister stop canister || true`, {
+                            stdio: 'inherit'
+                        });
+
+                        execSync(`dfx canister delete canister || true`, {
+                            stdio: 'inherit'
+                        });
+
+                        execSync(`dfx deploy canister`, {
+                            stdio: 'inherit'
+                        });
+
+                        const startingVer = 1n; // Canister starts at 0 when it's created but the deploy in pretest should increment it to be at 1 by the time it gets here
```

Comment:
But if you delete the canister, won't the version be entirely reset? I think the comment is just slightly incorrect is all, it's not pretest that increments but the deploy just above

Reply:
done